### PR TITLE
Add user_context to halide_opengl_output_client_bound()

### PR DESCRIPTION
**Pull request by steven-johnson**
Also clean up HelloAndroidGL to use HalideRuntimeOpenGL.h instead of
extern “C”.

### CODE REVIEW

File: src/runtime/opengl.cpp [hunk 1]
```diff
@@ -1953,7 +1953,7 @@ WEAK uintptr_t halide_opengl_get_texture(void *user_context, struct buffer_t *bu
 // indicating that the OpenGL object corresponding to the buffer_t is bound by
 // the app and not by the Halide runtime. For example, the buffer_t may be
 // backed by an FBO already bound by the application.
-WEAK uint64_t halide_opengl_output_client_bound() {
+WEAK uint64_t halide_opengl_output_client_bound(void *user_context) {
```
Comment (zvookin) [inline src/runtime/opengl.cpp]: You should comment out the argument name if it is not used.

**Comment by zvookin** [comment]
I guess I sort of see the point of this, but is this driven by something not working or just a clean up?

**Comment by zvookin** [comment]
I think my main issue here is that calling this with NULL user_context pretty much has to return the constant in question. user_context is so nebulous to start with that this will probably be a common occurrence. Now that we have device wrappers, the value used here is negotiated at the device interface level, not the GLSL one. So to do this properly, GLSL should be reserving a wrapper from device_interface instead of using a constant. In practice that really does make it context dependent, but passing NULL has to work...

**Comment by zvookin** [comment]
I guess the other way to go is to have this routine wrap the constant. This will require freeing the wrapper, but that should happen correctly in general anyway.

**Comment by steven-johnson** [comment]
It's just cleanup: I noticed that we were using it inconsistently in the AndroidGL app. 

If this is always going to return the constant in question, why do we need the runtime call?

**Comment by steven-johnson** [comment]
OK, clearly this has more issues, per Issue #774; closing this and rethinking

**Pull request closed without merging**